ev3/legoev3_ports_out.c

```diff
@@ -677,16 +677,33 @@ static int ev3_output_port_set_device(void *context, const char *device_name)
 {
 	struct ev3_output_port_data *data = context;
 	struct lego_device *new_motor;
+	struct ev3_motor_platform_data pdata;
+
+	switch (data->out_port.mode) {
+	case EV3_OUTPUT_PORT_MODE_TACHO_MOTOR:
+		if (data->motor_type != MOTOR_TACHO)
+			return -EOPNOTSUPP;
+
+		pdata.tacho_int_gpio = data->gpio[GPIO_PIN5_INT].gpio;
+		pdata.tacho_dir_gpio = data->gpio[GPIO_PIN6_DIR].gpio;
 
-	if (data->motor_type != MOTOR_TACHO)
+		break;
+	case EV3_OUTPUT_PORT_MODE_AUTO:
```

Comment:
Previously, you could still change the driver when in auto mode. The input port drivers also allow you to change the sensor driver when in auto mode. To keep things consistent, I think we should leave this the way it was.

We just need to keep the fix for platform data.